File: src/MyLiverpool.Web.WebApiNext/ClientApp/src/app/network/shared/contracts/upsert-contract.command.ts
import { BaseEntity } from '@domain/models';

export namespace UpsertContractCommand {

    export class Request extends BaseEntity<Request> {

        public salary: number;

        public personId: number;

        public startDate: Date;

        public endDate: Date;
    }

    export class Response extends BaseEntity<Response> {
        public id: number;
    }
}

File: src/MyLfc.Application/Features/Contracts/Commands/UpsertContractCommand.cs
using System;
using FluentValidation;

namespace MyLfc.Application.Features.Contracts.Commands;

public class UpsertContractCommand
{
    public abstract class Request
    {
        public int Salary { get; set; }

        public int PersonId { get; set; }

        public DateTimeOffset StartDate { get; set; }

        public DateTimeOffset EndDate { get; set; }
    }


    public abstract class Validator<T> : AbstractValidator<T> where T : Request
    {
        protected Validator()
        {

        }
    }
}
